### PR TITLE
Add “h100” as an alias for the "littleh" unit

### PR DESCRIPTION
**Pull request by nstarman**
Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

Request Review: @mhvk @eteq

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will
review this pull request of some common things to look for. This list
is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label.
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] If the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/units/astrophys.py [hunk 1]
```diff
@@ -123,9 +123,7 @@
 # This is not formally a unit, but is used in that way in many contexts, and
 # an appropriate equivalency is only possible if it's treated as a unit (see
 # https://arxiv.org/pdf/1308.4150.pdf for more)
-# Also note that h or h100 or h_100 would be a better name, but they either
-# conflict or have numbers in them, which is apparently disallowed
-def_unit(['littleh'], namespace=_ns, prefixes=False,
+def_unit(['littleh', 'h100'], namespace=_ns, prefixes=False,
```
Comment (pllim) [inline astropy/units/astrophys.py:126]: Huh, so numbers are allowed now?

Comment (nstarman) [reply astropy/units/astrophys.py:126]: I guess so. :).

Comment (mhvk) [reply astropy/units/astrophys.py:126]: @pllim is right that numbers should not be there, as they are normally interpreted as a power: 
```
In [2]: u.Unit('cm2')
Out[2]: Unit("cm2")

In [3]: u.Unit('cm2') == u.cm**2
Out[3]: True
```

Do try whether combinations like `u.Unit('h100 Mpc')` work.

Comment (nstarman) [reply astropy/units/astrophys.py:126]: Egads. That produces $hour^{100} * Mpc$

Comment (nstarman) [reply astropy/units/astrophys.py:126]: Well. Closing.

**Comment by nstarman** [comment]
Numbers still don't work in units. 😢

**Pull request closed without merging**